src/main/java/seedu/address/model/activity/Amount.java

```diff
@@ -26,7 +27,7 @@ public Amount(double amount) {
      * Returns true if the given amount is a valid amount.
      */
     public static boolean isValidAmount(double test) {
-        return test > 0;
+        return test > 0 && test <= 1000000 && test * 100 == (int) (test * 100);
```

Comment:
This looks like a pretty dodgy way to check if the valid amount is is specified to 2 d.p. - floating-point numbers may have different behaviour on different OSes (Windows for example, sometimes truncates trailing 0s from the decimal).

I think instead of limiting the decimal to exactly 2 d.p., we should just truncate/round the input to 2 d.p. (put this in the User Guide). Additionally, floating-point errors mean these two command strings behave very differently:
`expense e/25.50000000000001 p/Alex` - fails correctly
`expense e/25.500000000000001 p/Alex` - succeeds due to loss of precision
`expense e/999999.99999999999 p/Bernice` - succeeds due to loss of precision

Reply:
updated